### ISSUE TITLE
General questions

### ISSUE THREAD

**Issue by flav27** [snu-mllab/DCPG#1]
Thank you for making the code public !

I have several questions if you have the time for them:

1. _"We identify that the value network trained on multiple environments is more likely to
memorize the training data and cannot generalize to unvisited states of the training environments,
which acts detrimentally to not only the training performance but also the test performance on unseen
environments."_

- Does this mean that training with a smaller number of levels, lets say 100 instead of 200, will improve the test results ? I'm interested if you have a metric for it.

2. About the rollout buffer:

- `num_processes: 64
    num_steps: 256`

- I've seen that PPG, DAAC and now DCPG use this parameters, is this the sweet spot, or if the network is trained with more processes and for a longer time the test performance will improve ?
- For example:
`num_processes: 256
 num_steps: 256
  num_env_steps: 100M
 `

3. What is the intuition behind making the batch size from aux phase smaller than the batch size from ppo phase ?

4. From experience masking the invalid actions speed-up the training process.
    If I mask the invalid actions should I use also a distribution that supports masking where the dyna loss is calculated ?
    
    `rand_action_dists = Categorical(probs=probs)
     rand_actions_batch = rand_action_dists.sample().unsqueeze(dim=1)`

Thank you

**Comment by symoon11** [comment]
Thank you for your attention to our paper. Here is the answer to your question below.

Q1. To obtain an agent that has better test performance, we have to train the agent such that 1) the agent should have high training performance and 2) the generalization gap should be small. In this paper, we focus on training an agent that has high training performance in multiple training environments.

Q2. These hyper-parameter values are borrowed from the previous works (PPG, DAAC). Increasing the number of environment interactions indeed improves test performance.

Q3. We use the same hyper-parameter setting as PPG for a fair comparison. I think increasing the batch size for the aux phase 2x will not affect train and test performance.

Q4. If you want to constrain the action space, you should also sample random action from the constrained action space.
